EcosiaTests/WhatsNewLocalDataProviderTests.swift

```diff
@@ -102,12 +122,39 @@ final class WhatsNewLocalDataProviderTests: XCTestCase {
         User.shared.whatsNewItemsVersionsShown = ["9.0.0"]
         EcosiaInstallType.set(type: .upgrade)
         EcosiaInstallType.updateCurrentVersion(version: "8.0.0")
+        let items: [Version: [WhatsNewItem]] = [
+            Version("8.2.0")!: [.emptyItem()],
+            Version("9.0.0")!: [.emptyItem()]
+        ]
         let dataProvider = WhatsNewLocalDataProvider(versionProvider: MockAppVersionInfoProvider(mockedAppVersion: "9.0.2"))
         
         // When
         let shouldShowWhatsNew = dataProvider.shouldShowWhatsNewPage
         
         // Then
+        XCTAssertFalse(dataProvider.getVersionRange() == [Version("8.2.0")!], "The version 8.2.0 should not be among the picked version range. Resulting version range: \(dataProvider.getVersionRange())")
         XCTAssertFalse(shouldShowWhatsNew, "Upgrade with already shown items should show not show what's new")
     }
+    
+    func testUpgradeToVersionWithoutItemsLike940ShouldNotShowWhatsNew() {
```

Comment:
Can you clarify why the bug was happening? Was it just a matter of excluding fromIndex and including toIndex? Reading the code to me I still don't understand why it would happen only on a major version bump.

Reply:
Thanks for asking!
It wasn't only on a major version bump.
Mostly on a fresh app install of a version that doesn't have items to show.
Then upgrade to a version that (still) doesn't have items to show.
We are into an `.upgrade` state without having marked the items as shown. Hence for the logic there are items to be shown.

Reply:
Got it! Makes sense then, I guess it's a bit worse that I reported 🙈 Good that we are fixing it now 🎉